evalml/tests/automl_tests/test_automl.py

```diff
@@ -2088,3 +2089,87 @@ def test_automl_validate_objective(non_core_objective, X_y_regression):
     with pytest.raises(ValueError, match='is not allowed in AutoML!'):
         AutoMLSearch(X_train=X, y_train=y, problem_type=non_core_objective.problem_types[0],
                      additional_objectives=[non_core_objective.name])
+
+
+def test_automl_pipeline_params_simple(X_y_binary):
```

Comment:
Currently, these tests run AutoMLSearch to make sure that the parameters we specified are used. I noticed that it was harder to check for `best_pipeline` parameters, which is what I'm currently using to check, if I mock `.fit()` and `.score()`. If anyone has recommendations for how to potentially speed this up, that would be super helpful! (It currently takes ~9.23 seconds for the three new test functions I added)

Reply:
Can you explain the problem with mocking `fit` and `score`? I wouldn't go by `best_pipeline`. I'd mock `fit` and `score` and the analyze the `_results` dict or rankings table to make sure the parameters are within the new ranges.

Reply:
Right, my initial idea was that it would be nice to make sure the final parameters don't include parameters not in our hyperparam ranges, but if we initialize the pipelines correctly, we can assume the tuner will work properly, especially since we've already had tuner tests to cover this. This makes sense! I'll add in the mocking and loop through the results instead of looking at `best_pipeline`